#High-priority Order #4

#Number of orders: 2
#Number of shipments per order:
# -order_0:
#   -shipment_0
#   -shipment_1
# -order_1:
#   -shipment_0
#Challenges:
#  -Interruption: Yes
#  -Drop parts: Yes
#  -Sensor blackout: No
#  -Part flipping: Yes
#  -Faulty parts: Yes
#  -Moving obstacle: No
#Containers
# -Belt: Yes
# -Bins: No
# -Shelf: Yes
#AGV
# -order_0:
#   -shipment_0: agv1
#   -shipment_1: agv2
# -order_1:
#   -shipment_0: agv2
#Time limit: Yes

time_limit: 500

options:
  gazebo_state_logging: true
  insert_models_over_bins: false
  insert_models_over_shelves: true
  belt_population_cycles: 20
  spawn_extra_models: true
  model_type_aliases:
    order_part1: pulley_part_blue
    order_part2: gear_part_red
    order_part3: gasket_part_green
    order_part4: pulley_part_red

shelf_layout:
  row_1: ['base',0,'base','base'] #shelves: 3, 4, 5
  row_2: ['pipe','base',0,'base']   #shelves: 6, 7, 8
  row_3: ['base','pipe',0,'collar'] #shelves: 9, 10, 11


orders:
  order_0:
    announcement_condition: time
    announcement_condition_value: 0.0
    shipment_count: 2
    destinations: [agv1,agv2]
    products:
      part_0:
        type: pulley_part_red
        pose:
          xyz: [0.1, 0.1, 0]
          rpy: [0, 0, 0]
      part_1:
        type: pulley_part_blue
        pose:
          xyz: [-0.1, -0.1, 0]
          rpy: ['pi', 0, 0]
  order_1:
    announcement_condition: unwanted_products
    announcement_condition_value: 1
    shipment_count: 1
    destinations: [agv2]
    products:
      part_0:
        type: gear_part_red
        pose:
          xyz: [0.0, 0.0, 0]
          rpy: [0, 0, 0]
      part_1:
        type: gasket_part_green
        pose:
          xyz: [0.15, 0.2, 0]
          rpy: [0, 0, 'pi/4']




belt_models:
  pulley_part_blue:
    15.0:
      pose:
        xyz: [0.0, 4.3, 0.92]
        rpy: [0, 0, 'pi/2']
  # pulley_part_red:
  #   15.0:
  #     pose:
  #       xyz: [0.0, 4.3, 0.92]
  #       rpy: [0, 0, 'pi/2']


#2 arms advantageous
models_over_shelves:
  shelf5:
    models:
      gear_part_red:
        xyz_start: [1, 0.45, 0.0]
        xyz_end: [0., 0.45, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 6
        num_models_y: 1
      pulley_part_red:
        xyz_start: [1.5, -0.45, 0.0]
        xyz_end: [-0.5, -0.45, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 4
        num_models_y: 1
  shelf11:
    models:
      gasket_part_green:
        xyz_start: [1, 0.45, 0.0]
        xyz_end: [0., 0.45, 0.0]
        rpy: [0, 0, 'pi/4']
        num_models_x: 6
        num_models_y: 1

#Faulty gripper
drops:
  drop_regions:
    shipping_box_0_impeding:
      frame: agv1::kit_tray_1
      min:
        xyz: [-0.3, -0.3, 0.0]
      max:
        xyz: [0.3, 0.3, 0.5]
      destination:
        xyz: [0.2, 0.3, 0.05]
        rpy: [0, 0, 0.2]
      product_type_to_drop: pulley_part_red
    shipping_box_1_impeding:
      frame: agv2::kit_tray_2
      min:
        xyz: [-0.3, -0.3, 0.0]
      max:
        xyz: [0.3, 0.3, 0.5]
      destination:
        xyz: [0.15, 0.15, 0.05]
        rpy: [0, 0, 0.2]
      product_type_to_drop: gear_part_red
#Faulty products
faulty_products:
  - gear_part_red_1
  - gasket_part_green_2
